crates/relayer/src/chain/namada.rs

```diff
@@ -0,0 +1,1270 @@
+                let client_id =
+                    storage::client_id(&key).map_err(|e| Error::query(e.to_string()))?;
+                let client_id = ClientId::from_str(&client_id.to_string()).unwrap();
+                let client_state = AnyClientState::decode_vec(&value).map_err(Error::decode)?;
+                states.push(IdentifiedAnyClientState::new(client_id, client_state));
+            }
+        }
+
+        Ok(states)
+    }
+
+    fn query_client_state(
+        &self,
+        request: QueryClientStateRequest,
+        include_proof: IncludeProof,
+    ) -> Result<(AnyClientState, Option<MerkleProof>), Error> {
+        crate::time!(
+            "query_client_state",
+            {
+                "src_chain": self.config().id.to_string(),
+            }
+        );
+        crate::telemetry!(query, self.id(), "query_client_state");
+
+        let path = ClientStatePath(request.client_id);
+        let key = storage::ibc_key(path.to_string()).expect("the path should be parsable");
+        let (value, proof) = self.query(key, request.height, include_proof)?;
+        let client_state = AnyClientState::decode_vec(&value).map_err(Error::decode)?;
+
+        Ok((client_state, proof))
+    }
+
+    fn query_consensus_state(
+        &self,
+        request: QueryConsensusStateRequest,
+        include_proof: IncludeProof,
+    ) -> Result<(AnyConsensusState, Option<MerkleProof>), Error> {
+        crate::time!(
+            "query_consensus_state",
+            {
+                "src_chain": self.config().id.to_string(),
+            }
+        );
+        crate::telemetry!(query, self.id(), "query_consensus_state");
+
+        let path = ClientConsensusStatePath {
+            client_id: request.client_id,
+            epoch: request.consensus_height.revision_number(),
+            height: request.consensus_height.revision_height(),
+        };
+        let key = storage::ibc_key(path.to_string()).expect("the path should be parsable");
+        let (value, proof) = self.query(key, request.query_height, include_proof)?;
+        let consensus_state = AnyConsensusState::decode_vec(&value).map_err(Error::decode)?;
+        Ok((consensus_state, proof))
+    }
+
+    fn query_consensus_state_heights(
+        &self,
+        request: QueryConsensusStateHeightsRequest,
+    ) -> Result<Vec<ICSHeight>, Error> {
+        let prefix = storage::ibc_key(format!("clients/{}", request.client_id))
+            .expect("the path should be parsable");
+        let mut heights = vec![];
+        for prefix_value in self.query_prefix(prefix)? {
+            let PrefixValue { key, value: _ } = prefix_value;
+            match storage::consensus_height(&key) {
+                Ok(h) => {
+                    let height = ICSHeight::new(h.revision_number(), h.revision_height()).unwrap();
+                    heights.push(height);
+                }
+                // the key is not for a consensus state
+                Err(_) => continue,
```

Comment:
Is this expected to happen often?
If not, a `debug!` would be nice to have the error. If it happens often then using `trace!` would be preferred.

Reply:
added `debug!`